pages/search.tsx

```diff
@@ -66,101 +69,155 @@ export default function Search({
-                    })}
-                  >
-                    Showing {data.products.length} results{' '}
-                    {q && (
-                      <>
-                        for "<strong>{q}</strong>"
-                      </>
-                    )}
-                  </span>
-                  <span
-                    className={cn('animated', {
-                      fadeIn: !data.found,
-                      hidden: data.found,
-                    })}
-                  >
-                    {q ? (
-                      <>
-                        There are no products that match "<strong>{q}</strong>"
-                      </>
-                    ) : (
-                      <>
-                        There are no products that match the selected category &
-                        designer
-                      </>
-                    )}
-                  </span>
-                </>
-              ) : q ? (
-                <>
-                  Searching for: "<strong>{q}</strong>"
-                </>
-              ) : (
-                <>Searching...</>
+            <div className="mb-6 transition ease-in duration-75">
+              <div className="flex justify-between">
+                {data ? (
+                  <>
+                    <span
+                      className={cn('animated', {
+                        fadeIn: data.found,
+                        hidden: !data.found,
+                      })}
+                    >
+                      Showing {data.products.length} results{' '}
+                      {q && (
+                        <>
+                          for "<strong>{q}</strong>"
+                        </>
+                      )}
+                    </span>
+                    <span
+                      className={cn('animated', {
+                        fadeIn: !data.found,
+                        hidden: data.found,
+                      })}
+                    >
+                      {q ? (
+                        <>
+                          There are no products that match "<strong>{q}</strong>"
+                        </>
+                      ) : (
+                        <>
+                          There are no products that match the selected category &
+                          designer
+                        </>
+                      )}
+                    </span>
+                  </>
+                ) : q ? (
+                  <div>
+                    Searching for: "<strong>{q}</strong>"
+                  </div>
+                ) : (
+                  <>Searching...</>
+                )}
+
+                <button
+                  className="md:invisible text-accents-8 outline-none"
+                  onClick={() => setShowMobileMenu(prev => !prev)}
+                >
+                  Filter & Sort
+                </button>
+              </div>
+
+              {showMobileMenu && (
```

Comment:
Hi there, my 2 cents is, why you choose use `showMobileMenu = true` to show mobile content and not use `showMobileMenu = false` to show desktop content? I see a duplicated `<Categories /> <Designers /> and <Sort />` elements on the final markup.

Reply:
Actually, there are two "menus" now, one appears on `md+` devices and another on `md-` devices. This behavior is controlled by the tailwind classes, that's why the duplicated `<Categories /> <Designers /> and <Sort />`.

The `showMobileMenu` state is used to control only the click on the "Filter & Sort" button.

Reply:
Exactly because there are two identical menus but with different views, I said it could be refactored to use the variable when it is mobile or not, thus saving code.